### PR TITLE
Add jenkins.plugin.withWarning metric

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?

Adds new metric `jenkins.plugin.withWarning` which is the count of currently installed plugins that have warnings.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogJenkinsPublisher.java [hunk 1]
```diff
@@ -122,4 +133,16 @@ private PluginData collectPluginData(Jenkins instance) {
 
         return pluginData.build();
     }
+
+    @SuppressRestrictedWarnings({ UpdateSiteWarningsMonitor.class, UpdateSite.Warning.class })
+    private Integer getPluginWarningsCount() {
+        try {
+            UpdateSiteWarningsMonitor updateSiteWarningsMonitor = ExtensionList.lookupSingleton(UpdateSiteWarningsMonitor.class);
+            Map<PluginWrapper, List<UpdateSite.Warning>> activePluginWarningsByPlugin = updateSiteWarningsMonitor.getActivePluginWarningsByPlugin();
+            return activePluginWarningsByPlugin.size();
```
Comment (drodriguezhdez) [inline src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogJenkinsPublisher.java:142]: Do we know if this is thread safe?

File: pom.xml [hunk 2]
```diff
@@ -210,6 +210,11 @@
       <artifactId>support-log-formatter</artifactId>
       <version>1.2</version>
     </dependency>
+    <dependency>
+      <groupId>org.kohsuke</groupId>
+      <artifactId>access-modifier-suppressions</artifactId>
+      <version>${access-modifier-checker.version}</version>
+    </dependency>
```
Comment (drodriguezhdez) [inline pom.xml:217]: Can this new dependency enters in conflict with other deps from Jenkins?

**Review by drodriguezhdez — COMMENTED** [review]
Dropped some comments

File: pom.xml (hunk 2 above)
Comment (nikita-tkachenko-datadog) [reply pom.xml:217]: The version comes from Jenkins BOM, which is there to ensure that different dependencies that the plugin has are mutually compatible.

As for inter-plugin conflicts, the [docs](https://www.jenkins.io/doc/developer/plugin-development/dependencies-and-class-loading/) say that each plugin has a dedicated classloader at runtime, so I don't think there'll be any issues.

File: src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogJenkinsPublisher.java (hunk 1 above)
Comment (nikita-tkachenko-datadog) [reply src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogJenkinsPublisher.java:142]: I checked the `getActivePluginWarningsByPlugin()` method's implementation, and the map that it returns is local, so there's no other thread that can reach it.

**Pull request merged**